src/widgets/survey/ui/FlowSurvey.tsx

```diff
@@ -59,13 +61,18 @@ function FlowSurvey({
     onClose();
   }
 
-  function complete() {
+  function complete(
+    reason: 'regular' | 'idle' = 'regular',
+    activitySummary: ActivitySummaryData | null = null,
+  ) {
     const isLast = step === pipeline.length - 1;
 
     if (isLast) {
       closeFlow();
+    } else if (reason === 'idle') {
+      idleTimeoutNext();
     } else {
-      next();
+      next(activitySummary);
```

Comment:
Let's save the summary right inside Intermediate and Finish components with `useFlowActions`